### ISSUE TITLE
About freq_decompose

### ISSUE THREAD

**Issue by foureye** [VITA-Group/ViT-Anti-Oversmoothing#8]
Hi，the paper provides a novel analysis of the low-pass filtering of Transformers self-attention matrix from the frequency domain and includes extensive visualization.I have some questions about freq_decompose.
featscale.py 42-45
def freq_decompose(self, x):
        x_d = torch.mean(x, -2, keepdim=True) # [bs, 1, dim]
        x_h = x - x_d # high freq [bs, len, dim]
        return x_d, x_h
when decompose the self-attention matrix,I have not found any code about frequency domain transforming，such as torch.FFT or DFT matrix.The process of freq_decompose is still on image space but can realize the decompose of DC and high component,or I miss the code of the frequency domain transforming?
Thank you!

**Comment by peihaowang** [comment]
Hi,

Thanks for your interest. You are right. Although the `freq_decompose` function is implemented on the image space, it realizes Fourier domain operations. It is noteworthy that the DC component of signals is simply the mean of the whole signal. So you don't need to run Fourier transform to extract this information. Nevertheless, if you consider a more complicated Fourier analysis, you may need `torch.FFT` then. 

Best,
Peihao
